x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/rate/InternalResetTrackingRate.java

```diff
@@ -109,7 +126,10 @@ public XContentBuilder doXContentBody(XContentBuilder builder, Params params) th
 
     @Override
     public double value() {
-        return (endValue - startValue + resetCompensation) / (endTime - startTime);
+        long rateUnitSeconds = rateUnit == null
+            ? Rounding.DateTimeUnit.SECOND_OF_MINUTE.getField().getBaseUnit().getDuration().getSeconds()
+            : rateUnit.getField().getBaseUnit().getDuration().toSeconds();
+        return (endValue - startValue + resetCompensation) / (endTime - startTime) * rateUnitSeconds;
```

Comment:
I don't think I fully understand this. Both `endTime` and `startTime` are millis based. How is multiplying the difference between `endTime` and `startTime` in millis with number of seconds of the duration do the conversion the specified unit? Looking at `AbstractRateAggregator` the divisor boils down to (in one of the `PreparedRounding` implementations): `nextUtcMillis - utcMillis` / number_of_millis_of_unit` in case unit is millis based and parent bucket is date histogram. Shouldn't this code not do: `... / ((endTime - startTime) / rateUnitMillis)`?

Reply:
I think is the same...I did a simplification... (a/ (b / c)) = a / b * 1 / (1 / c) = a / b * c right?

Reply:
Salvatore and I chatted and this is a simplification of:

```
long rateUnitMillis = rateUnit == null
            ? Rounding.DateTimeUnit.SECOND_OF_MINUTE.getField().getBaseUnit().getDuration().toMillis()
            : rateUnit.getField().getBaseUnit().getDuration().toMillis();
        return (endValue - startValue + resetCompensation) / (endTime - startTime) * rateUnitMillis / 1000;
```